core/providers/java/java.go

```diff
@@ -0,0 +1,87 @@
+package java
+
+import (
+	"fmt"
+
+	"github.com/railwayapp/railpack/core/generate"
+	"github.com/railwayapp/railpack/core/plan"
+)
+
+type JavaProvider struct{}
+
+func (p *JavaProvider) Name() string {
+	return "java"
+}
+
+func (p *JavaProvider) Detect(ctx *generate.GenerateContext) (bool, error) {
+	return ctx.App.HasMatch("pom.{xml,atom,clj,groovy,rb,scala,yaml,yml}") || ctx.App.HasMatch("gradlew"), nil
+}
+
+func (p *JavaProvider) Initialize(ctx *generate.GenerateContext) error {
+	return nil
+}
+
+func (p *JavaProvider) StartCommandHelp() string {
+	return ""
+}
+
+func (p *JavaProvider) Plan(ctx *generate.GenerateContext) error {
+
+	build := ctx.NewCommandStep("build")
+	build.AddCommand(plan.NewCopyCommand("."))
+	build.Inputs = []plan.Input{plan.NewStepInput(ctx.GetMiseStepBuilder().Name())}
+
+	if p.usesGradle(ctx) {
+		p.setGradleVersion(ctx)
+		p.setJDKVersion(ctx, nil)
```

Comment:
Instead of passing in `nil`, can you always pass in the mise step that should be used so it is clear.